python/thunder/utils/datasets.py

```diff
@@ -63,6 +63,44 @@ def generate(self, k=3, npartitions=10, nrows=100, ncols=10, seed=None):
         else:
             return data
 
+class FAData(DataSets):
+
+    def generate(self, q=1, p=3, nrows=50, npartitions=10, sigmas=None, seed=None):
+        """
+        Generate data from a factor analysis model
+
+        Parameters
+        ----------
+        q : int
```

Comment:
Indicate the default, here and elsewhere, as in `q : int, optional, default = 1`